src/waitress/task.py

```diff
@@ -244,8 +249,8 @@ def close_on_finish():
                     response_headers.append(("Transfer-Encoding", "chunked"))
                     self.chunked_response = True
 
-                if not self.close_on_finish:
-                    close_on_finish()
+                if not self.self.set_close_on_finish:
```

Comment:
`self.self`?

This change likely should not exist, it should still be `if not self.close_on_finish` and then call `self.set_close_on_finish()`.

This seems like a search/replace failure?

Reply:
Thanks!